spec/models/report_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+RSpec.describe Report, type: :model do
+  let(:user){
+    FactoryBot.create :user
+  }
+  let(:user_1){
+    FactoryBot.create :user
+  }
+  let(:user_2){
+    FactoryBot.create :user
+  }
+  let(:department){
+    FactoryBot.create :department
+  }
+
+  let(:department_1){
+    FactoryBot.create :department
+  }
+
+  let(:department_2){
+    FactoryBot.create :department
+  }
+  let(:report){
+    FactoryBot.create :report
+  }
+
+  let(:report_1){
+    FactoryBot.create :report, user_id: user_1.id, department_id: department_1.id
+  }
+
+  let(:report_2){
+    FactoryBot.create :report, user_id: user_2.id, department_id: department_2.id
+  }
+
+  describe "presence" do
+    it { is_expected.to validate_presence_of(:plan_today) }
+    it { is_expected.to validate_presence_of(:reality) }
+    it { is_expected.to validate_presence_of(:plan_next_day) }
+  end
+
+  describe "Associations" do
+    it { is_expected.to belong_to(:user) }
+    it { should belong_to(:department).optional(true) }
+    it { should belong_to(:user_department).optional(true) }
+  end
+
+  describe "Validations" do
+    context "when field plan today" do
+      subject{FactoryBot.build(:report)}
+      it {should validate_presence_of(:plan_today)}
+      it {is_expected.to validate_length_of(:plan_today).is_at_most(Settings.report.max_length)}
+    end
+
+    context "when field reality" do
+      subject{FactoryBot.build(:report)}
+      it {should validate_presence_of(:reality)}
+      it {is_expected.to validate_length_of(:reality).is_at_most(Settings.report.max_length)}
+    end
+
+    context "when field plan next day" do
+      subject{FactoryBot.build(:report)}
+      it {should validate_presence_of(:plan_next_day)}
+      it {is_expected.to validate_length_of(:plan_next_day).is_at_most(Settings.report.max_length)}
+    end
+  end
+
+  describe "check scope" do
+    context "orders by created_at descending" do
```

Comment:
phân biệt describe và context

Reply:
- describe dùng để nhóm các test case theo class, module, method hoặc theo action của controller.
- context dùng để nhóm các điều kiện của example.
- it được sử dụng để định nghĩa một "Ví dụ" hay nói cách khác là dùng để định nghĩa 1 test case.